### PR TITLE
Script update for 1880 issue

### PR DESCRIPTION
**Pull request by GetmanetsIrina**
ATF Test Scripts to check #[FORDTCN-11972](https://luxproject.luxoft.com/jira/browse/FORDTCN-11972)

This PR is **ready for internal review** for review.

### Summary
Script update for 1880 issue

### ATF version
develop

### Changelog
- Fixed wrong time calculation for PI
- Added case without timeout for PI

 #### Note: 
Script is failed on develop branch, because SDL adds `DefaultTimeoutCompensation` for each part of timeout for PI (for default value, for UI part and for VR). My assumption that this is SDL issue according to `DefaultTimeoutCompensation` description (https://github.com/smartdevicelink/sdl_core/blob/65947fd1eb30a392948d67003df0fe803c070c6b/src/appMain/smartDeviceLink.ini#L61)
SDL issue will be created after internal review is finished.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/4_5/1880_DefaultAndRPCTimeoutGENERIC_ERROR.lua [hunk 1]
```diff
@@ -11,12 +11,15 @@ local commonDefects = require("test_scripts/Defects/4_5/commonDefects")
 local apiLoader = require("modules/api_loader")
 local api = apiLoader.init("data/MOBILE_API.xml")
 local schema = api.interface[next(api.interface)]
+local actions = require("user_modules/sequences/actions")
 
 --[[ Test Configuration ]]
 runner.testSettings.restrictions.sdlBuildOptions = { { extendedPolicy = { "PROPRIETARY", "EXTERNAL_PROPRIETARY" } } }
```
Comment (VjKlepikov) [inline test_scripts/Defects/4_5/1880_DefaultAndRPCTimeoutGENERIC_ERROR.lua]: I propose to extend coverage for HTTP mode

File: test_scripts/Defects/4_5/1880_DefaultAndRPCTimeoutGENERIC_ERROR.lua [hunk 2]
```diff
@@ -313,17 +351,17 @@ local function PerformInteraction(params, self)
   local RespTimeout
   if params.timeout then
     PIDuration = params.timeout
-    RespTimeout = DefaultTimeout + 2*PIDuration
+    RespTimeout = GeneralTimeout + 2*PIDuration
```
Comment (VjKlepikov) [inline test_scripts/Defects/4_5/1880_DefaultAndRPCTimeoutGENERIC_ERROR.lua]:
```suggestion
    RespTimeout = GeneralTimeout + PIDuration
```
I believe that the double timeout is redundant

**Review by VjKlepikov — CHANGES_REQUESTED** [review]

File: test_scripts/Defects/4_5/1880_DefaultAndRPCTimeoutGENERIC_ERROR.lua (hunk 1 above)
Comment (GetmanetsIrina) [reply test_scripts/Defects/4_5/1880_DefaultAndRPCTimeoutGENERIC_ERROR.lua]: Script calls specific PTU function that is not applicable for `HTTP` flow. Is is the reason why `HTTP` flow is not covered.
Update of this call will be the significant update, so propose to leave current implementation as is.

File: test_scripts/Defects/4_5/1880_DefaultAndRPCTimeoutGENERIC_ERROR.lua (hunk 2 above)
Comment (GetmanetsIrina) [reply test_scripts/Defects/4_5/1880_DefaultAndRPCTimeoutGENERIC_ERROR.lua]: Please find update in [a92da80](https://github.com/LuxoftSDL/sdl_atf_test_scripts/pull/174/commits/a92da8016cffbf484eb493195cf5404c3dd70ce2)

**Comment by GetmanetsIrina** [comment]
@VjKlepikov, please find update in [e1b399c](https://github.com/LuxoftSDL/sdl_atf_test_scripts/pull/174/commits/e1b399cffec37de2c236d8a9a769304576991e66)

File: test_scripts/Defects/4_5/1880/1880_2_PerformInteraction_BOTH_DefaultAndRPCTimeoutGENERIC_ERROR.lua [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+--  SDL must respond 'GENERIC_ERROR, success:false' to mobile app
+--
+-- Steps:
+-- 1. HMI and SDL are started
+-- 2. Mobile app is registered and activated
+-- 3. CreateInteractionChoiceSet is added
+-- 4. Mobile app requests PerformInteraction(BOTH)
+-- SDL does:
+-- - request VR.PerformInteraction and UI.PerformInteraction to HMI
+-- - start <DefaultTimeout> + <*RPCs_own_timeout> for expecting VR response
```
Comment (VjKlepikov) [inline test_scripts/Defects/4_5/1880/1880_2_PerformInteraction_BOTH_DefaultAndRPCTimeoutGENERIC_ERROR.lua]:
```suggestion
-- - start timeout <DefaultTimeout> + <*RPCs_own_timeout> for expecting VR response
```

**Review by VjKlepikov — CHANGES_REQUESTED** [review]
I propose to replace EXPECT_HMICALL -> actions.hmi.getConnection():ExpectRequest

File: test_scripts/Defects/4_5/1880/common.lua [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+  for param, value in pairs(params) do
+    mainParams[param] = value
+  end
+
+  PIDuration = params.timeout or getDefaultValueFromAPI("PerformInteraction", "timeout")
+  RespTimeout = GeneralTimeout + PIDuration + additionalTimeout
+  local cid = actions.mobile.getSession():SendRPC("PerformInteraction", mainParams)
+  RequestTime = timestamp()
+  EXPECT_HMICALL("VR.PerformInteraction")
```
Comment (VjKlepikov) [inline test_scripts/Defects/4_5/1880/common.lua]:
```suggestion
  actions.hmi.getConnection():ExpectRequest("VR.PerformInteraction")
```

**Review by VjKlepikov — CHANGES_REQUESTED** [review]

File: test_scripts/Defects/4_5/1880/common.lua [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+--[[ Module ]]
+local m = {}
+
+--[[ Proxy Functions ]]
+m.preconditions = actions.preconditions
+m.start = actions.start
+m.registerNoPTU = actions.app.registerNoPTU
+m.activate = actions.app.activate
+m.postconditions = actions.preconditions
```
Comment (VjKlepikov) [inline test_scripts/Defects/4_5/1880/common.lua]:
```suggestion
m.postconditions = actions.postconditions
```

**Review by VjKlepikov — CHANGES_REQUESTED** [review]

**Comment by GetmanetsIrina** [comment]
@VjKlepikov , please review

**Review by VjKlepikov — APPROVED** [review]

**Comment by GetmanetsIrina** [comment]
moved to open

**Pull request closed without merging**